content/blog/2023/2023-05-18-operators-in-typescript.md

```diff
@@ -0,0 +1,177 @@
+TypeScript is a superset of JavaScript that adds static typing and other features to the language. Its operators are crucial to understanding the language and writing effective code. Operators are symbols or keywords in a programming language that perform operations on values, such as arithmetic operations, string concatenation, and comparisons. **Understanding operators in TypeScript is essential because they are fundamental building blocks of the language and are used in almost every programming aspect.** By choosing the right operator for the job, you can often simplify your code and make it easier to understand and maintain. In this article, we will explore the most important operators in TypeScript and provide examples of how they can be used in real-world applications to help you write more efficient and readable code.
+
+{{% github "https://github.com/Irtaza2009/code-examples/tree/operators-in-typescript/nodejs/operators-in-typescript" %}}
+
+# What operators are in Typescript? How are they used?
+In Typescript, **operators are symbols used to perform operations on variables or values.** They can be classified into several categories based on their functions.
+
+## Concatenation Operators
+Concatenation operators in TypeScript are used to combine strings and values together. The most common concatenation operator is the plus sign (+). When used with strings, the plus sign combines them into a single string. When used with a string and a value, the plus sign concatenates the value to the end of the string.
+
+For example, let's say we have two strings, "Hello" and "World". We can use the concatenation operator to combine them into a single string:
+
+```ts
+let greeting = "Hello" + "World";
+
+console.log(greeting);
+```
+Output:
+~~~ts
+HelloWorld
+~~~
+
+We can also use the concatenation operator to combine a string and a value:
+
+~~~ts
+let age = 30;
+let message = "I am " + age + " years old.";
+console.log(message); 
+~~~
+Output:
+~~~ts
+I am 30 years old.
+~~~
+
+In this example, the concatenation operator combines the string "I am " with the value of the age variable (30) and the string " years old." to create the final message.
+
+Concatenation operators are useful in situations where we need to build dynamic strings based on values or user input. By using concatenation operators, we can create custom messages or outputs that are tailored to our specific needs.
+
+## Arithmetic Operators
```

Comment:
How do arithmetic operators work on non-number variables? Does the TypeScript compiler allow that? If yes, what's the result?